### PR TITLE
Plot circuit layout on backend coupling map

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
Adds `plot_circuit_layout` that plots the layout of a circuit on a backend coupling map.

Screen Shot 2019-07-22 at 3 51 02 PM

### CODE REVIEW

File: qiskit/visualization/gate_map.py [hunk 1]
```diff
@@ -114,6 +116,14 @@ def plot_gate_map(backend, figsize=None,
     n_qubits = config.n_qubits
     cmap = config.coupling_map
 
+    if qubit_labels is None:
+        qubit_labels = list(range(n_qubits))
+    else:
+        if len(qubit_labels) != n_qubits:
+            raise QiskitError('Length of qubit labels '
```
Comment (mtreinish) [inline qiskit/visualization/gate_map.py]: I would probably use a `VisualizationError` for this

File: qiskit/visualization/gate_map.py [hunk 2]
```diff
@@ -198,11 +208,72 @@ def plot_gate_map(backend, figsize=None,
...
+
+    Returns:
+        Figure: A matplotlib figure showing layout.
+
+    Raises:
+        Exception: Circuit has no layout attribute.
+        ValueError: Invalid view type given.
+    """
+    if circuit.layout is None:
+        raise Exception('Circuit has no layout. '
```
Comment (mtreinish) [inline qiskit/visualization/gate_map.py]: We probably should use a different exception from the base Exception here. I'd probably either use AttributeError or QiskitError.

File: qiskit/visualization/gate_map.py (hunk 1 above)
Comment (nonhermitian) [reply qiskit/visualization/gate_map.py]: Done.

File: qiskit/visualization/gate_map.py (hunk 2 above)
Comment (nonhermitian) [reply qiskit/visualization/gate_map.py]: And done.

**Pull request merged**